### ISSUE TITLE
gen_release_notes documentation sets GIT_TOKEN but code requires GITHUB_TOKEN

### ISSUE THREAD

**Issue by ghost** [Nebulaworks/orion#9]
The [user guide says to set `GIT_TOKEN`](https://github.com/Nebulaworks/orion/tree/main/scripts/gen_release_notes#how-to-use), but there is a [check in the code](https://github.com/Nebulaworks/orion/blob/main/scripts/gen_release_notes/git_parser/git_parser.py#L14-L17) for `GITHUB_TOKEN`.

**Comment by adam-nebulaworks** [comment]
A question this raises is does this script only support GitHub repos? Will it work on GitLab, or local-only clones? Might be worth calling out any requirements more clearly in the docs.

**Comment by jshcmpbll** [comment]
Ill take this one. Doing some digging to understand the differences between the two to see if we can make it more agnostic.

**Comment by PhinnyG** [comment]
Looking through the code it seems that this is currently set up to only handle GitHub repos because it makes calls to the GitHub API. (See here: https://github.com/Nebulaworks/orion/blob/main/scripts/gen_release_notes/git_parser/git_parser.py#L42-L49 ). 

Seems for now that editing the docs and making this clear is the best next step and then spin up a new issue for potentially making it generalized.

**Comment by adam-nebulaworks** [comment]
@PhinnyG yeah that scope limit makes sense. This issue is really about correcting a typo in the docs. I think it makes sense to just fix that and add a sentence or so clarifying that this only works with GitHub repos.

Also agreed on making a new issue. I'd be interested to see if we can make this work just on the local clone. If it only uses the API to get a list of changed files, we should be able to get the same outcome just by diffing the base and PR branches (which would have to be given as args or something). It'd be nice if we didn't need any access tokens. Ability to pull should be enough.
